hocs/ProtectedRoutes.tsx

```diff
@@ -0,0 +1,28 @@
+import { protectedRoutes, unprotectedRoutes } from "@/constants/const";
```

Comment:
No no no
This would've made sense if this were client side rendering
You have the option to check user auth on the server via `getServerSideProps` - utilise that
No HOCs needed